docs/side_quests/working_with_files.md

```diff
@@ -0,0 +1,1069 @@
+
+executor >  local (8)
+[d5/441891] process > ANALYZE_READS (sampleC) [100%] 8 of 8 ✔
+```
+
+Check the results directory now:
+
+```console title="Results Directory"
+> tree results
+results
+├── sampleA
+│   └── sampleA_stats.txt
+├── sampleB
+│   └── sampleB_stats.txt
+└── sampleC
+    └── sampleC_stats.txt
+```
+
+See how we have analyzed all the samples in one go!
+
+Wait, we have a problem. We have 2 replicates for sampleA, but only 1 output file! We are overwriting the output file each time.
+
+### 5.4. Make the published files unique
+
+Since we have access to the sample metadata, we can use it to make the output files unique.
+
+=== "After"
+
+    ```groovy title="main.nf" linenums="4"
+    publishDir "results/${meta.type}/${meta.id}/${meta.replicate}", mode: 'copy'
+    ```
+
+=== "Before"
+
+    ```groovy title="main.nf" linenums="4"
+    publishDir "results/${id}", mode: 'copy'
+    ```
+
+We have grabbed the metadata from the samples and used it to construct an output directory for each sample.
+
+Run the pipeline now and see all the results. Remove the results directory first to give yourself a clean workspace:
+
+```bash
+rm -r results
+nextflow run main.nf
+```
+
+```console title="Results Directory"
+ N E X T F L O W   ~  version 24.10.5
+
+Launching `./main.nf` [insane_swartz] DSL2 - revision: fff18abe6d
+
+executor >  local (8)
+[e3/449081] process > ANALYZE_READS (sampleC) [100%] 8 of 8 ✔
+```
+
+Check the results directory now:
+
+```console title="Results Directory"
+> tree results
+results/
+├── normal
+│   ├── sampleA
+│   │   ├── 1
+│   │   │   └── sampleA_stats.txt
+│   │   └── 2
+│   │       └── sampleA_stats.txt
+│   ├── sampleB
+│   │   └── 1
+│   │       └── sampleB_stats.txt
+│   └── sampleC
+│       └── 1
+│           └── sampleC_stats.txt
+└── tumor
+    ├── sampleA
+    │   ├── 1
+    │   │   └── sampleA_stats.txt
+    │   └── 2
+    │       └── sampleA_stats.txt
+    ├── sampleB
+    │   └── 1
+    │       └── sampleB_stats.txt
+    └── sampleC
+        └── 1
+            └── sampleC_stats.txt
+```
+
+See how using sample metadata as values gives us powerful flexibility in our pipeline. By propagating metadata alongside our data in tuples, we can:
+
+1. Create organized output directories based on sample attributes
+2. Make decisions in processes based on sample properties
+3. Split, join, and recombine data based on metadata values
+
+This pattern of keeping metadata explicit and attached to the data (rather than encoded in filenames) is a core best practice in Nextflow that enables building robust, maintainable bioinformatics workflows.
+
+### Takeaway
+
+- Processes can take files as input and produce new files as output
+- The `publishDir` directive organizes outputs based on metadata values
+- Metadata in tuples enables structured organization of results
+- This approach creates maintainable workflows with clear data provenance
+
+---
+
+## 6. Using remote files
```

Comment:
As a said further up, I don't think this should be a 'by the way' at the end, but rather something we show up-front.